### ISSUE TITLE
Automatic activation?

### ISSUE THREAD

**Issue by jagthedrummer** [ember-cli/ember-cli-deploy#73]
It would be nice to have a config option to automatically activate a new `index.html` whenever it is deployed. Would that be the kind of thing that should be an option in `ember-cli-deploy` itself, or should that be an add-on of some sort?

**Comment by achambers** [comment]
Hi @jagthedrummer.  This is something we are discussing but have not decided on the way forward as yet. We are going to be changing the way adapters work in ember-cli-deploy so we will be taking this in to account.

**Comment by jagthedrummer** [comment]
Cool, I'll be interested to see what's decided.

**Comment by lukemelia** [comment]
@jagthedrummer The goal we discussed  is that by default `ember deploy [target]` would fully deploy and activate by default, but that behavior would be configurable.

**Comment by HusseinMorsy** [comment]
@lukemelia that would be very useful

**Comment by EmergentBehavior** [comment]
I currently have ember-cli-deply as part of an Ansible task that automatically deploys the new index. Right now I have to manually dig up the short-hash for the new index, so this would be a welcome change :)

**Comment by achambers** [comment]
So, I'm looking in to this one.  And it got me wondering.  In order to run `activate` after deploying the index.html we need to know the key that the index was uploaded with.  This would mean that we would somehow need to pass the key from the `DeployIndexTask` into the `ActivateTask`.

However, as per the [ember-cli ARCHITECTURE.md](https://github.com/ember-cli/ember-cli/blob/master/ARCHITECTURE.md#tasks), tasks should **not have a return value by design**.  Technically, they should be resolving with `undefined`.

With that in mind, our `DeployIndexTask` already breaks this convention by returning the upload `key`, which makes me feel a little bit dirty.

So, technically, we could do this auto `activate` with the existing `DeployIndexTask`.  However, that task isn't really playing by the rules of an ember-cli `Task`.

Thoughts?

**Comment by justinko** [comment]
@EmergentBehavior 

> manually dig up the short-hash for the new index

Got a gist for that? ;)

**Comment by EmergentBehavior** [comment]
@justinko Sure! It's pretty straightforward. The relevant item from my `main.yml` in my `static_pipeline` role looks like this:

```YAML
- name: Retrieve 7-character short hash of last index commit for ember-cli-deploy
  local_action: command chdir="{{ my_project.client.path }}" git rev-parse --short=7 HEAD
  register: hash_result
  when: my_project.client.activate_index
```

The above is run after another command which runs `ember deploy --environment {{ some_environment}}`. The `{{my_project.client.path` above is the local path of my EmberJS project. This command just gets the short-hash of the last commit, saves it in `hash_result` and then I can use it later when activating the index on my remote Redis instance.

**Comment by justinko** [comment]
@EmergentBehavior ah! I thought you were regex'ing the output of `ember deploy` or `ember deploy:list`

Thanks!

**Comment by jrowlingson** [comment]
@EmergentBehavior 

```
ember deploy --environment staging && ember deploy:activate --revision=`ember deploy:list | grep '1)' | awk {'print $2'}`
```

**Comment by lukemelia** [comment]
FYI, in 0.5.0, you can accomplish this with `ember deploy staging --activate`

**Comment by HusseinMorsy** [comment]
:clap:

**Comment by jagthedrummer** [comment]
Hooray! Thanks to everyone involved!

**Comment by achambers** [comment]
@jagthedrummer Are you happy for me to close this issue as this functionality is available in v0.5.0-beta.1?

**Comment by jagthedrummer** [comment]
I'll save you the trouble!
